### ISSUE TITLE
Proper NzbDrone CDH integration for Usenet and Torrents

### ISSUE THREAD

**Issue by Taloth** [clinton-hall/nzbToMedia#583]
@clinton-hall 
I've been thinking about how to do a proper integration. Basically throwing out everything we've thought up before. No half measures this time.

The biggest problem is if you do transcoding and stuff to a directory that NzbDrone is unaware of, there's really no clean way to do that coz not every download client allows you to change the path.
Somehow we have to let nzbToMedia tell drone "Forget anything the download client api is telling you atm about this download, here is the directory you need, import it.... plz".

So I thought, why not send the download id along in the command. For sabnzbd it's the nzoid, for torrents it's the uppercase infohash, for nzbget it's a custom "drone" parameter we add to nzbget, or if that's unavailable nzbget's own Id property.

We may have to change how the command is handled in drone, coz atm the subsystem that actually processes the command is entirely unaware of download clients. Probably have to talk to markus about what command to create. Or just move the command handling logic elsewhere. But that's just Drone internals.

Atm Drone doesn't have an explicit return code, I think we should add that.

Anyway, lets check the scenarios:

**CDH disabled. Drone Factory.**   
-  Someone invokes the DownloadedEpisodesScanCommand with a path (and id).  
-  Drone triggers the Drone Factory scan and imports the episode.

**CDH enabled. Drone Factory.**   
-  Someone invokes the DownloadedEpisodesScanCommand with a path that resides inside Drone Factory without a downloadclientid.  
-  Drone triggers the Drone Factory and imports the episode.

**CDH enabled. Wrong path.**   
-  Someone invokes the DownloadedEpisodesScanCommand with a path not residing inside drone factory.  
-  Drone detects CDH is enabled and rejects anything that both has no id and has a path outside the drone factory.

**CDH enabled. Success.**   
-  NzbToMedia invokes the DownloadedEpisodesScanCommand with a path and id.  
-  Drone should detect CDH is enabled and use the id to find the related details.  
-  Drone will use the CDH logic to import from the specified directory and will mark the CDH item as imported if successful.

**CDH enabled. Failure.**
-  NzbToMedia invokes the DownloadedEpisodesScanCommand with a path and id.  
-  Drone should detect CDH is enabled and use the id to find the related details.  
-  Drone fails to import for some file naming issue.  
-  CDH would normally retry every minutes allowing the user to fix it, but since we don't have the correct path, we can't use the retry logic in this scenario. _Something to investigate_)  
-  Consequently, nzbtoMedia will report failure, which triggers NzbDrone Failed Download Handling etc etc.

**CDH enabled. Torrent client. Success.**
-  NzbToMedia invokes the DownloadedEpisodesScanCommand with a path and id. Drone should detect CDH is enabled and use the id to find the related details.  
-  Drone will use the CDH logic to import from the specified directory and will mark the CDH item as imported if succesful.  
- (Note that drone will have to 'move' stuff instead of copy or hardlink, leaving it the responsibility of NzbToMedia to hardlink)
-  NzbToMedia detects success and cleans up some files and subsequently returns success and the download client marks successful post-processing.
-  CDH will see the download as already imported and leaves the download alone until seeding finishes, after which (if enabled) the CDH Remove process is triggered and deletes the file and torrent from the download client.

Anyway to recap:  
_(not implemented, just an idea)_
Endpoint: http://localhost:8989/api/command
Request Header: X-Api-Key: {ApiKey}
Json Request: 

``` json
{
   "name": "downloadedepisodesscan",
   "path": "{target path to import}",
   "downloadClientId": "{id of download item}" ; Required for CDH otherwise optional.
}
```

Json Response:

``` json
{
   "success": true | false,
   "message": "{error message}", ; Optional
}
```

Lemme know what you think and whether you can think of a scenario that would not be handled properly.
